src/compiler/compile/render_dom/wrappers/Element/Binding.ts

```diff
@@ -268,10 +268,13 @@ function get_event_handler(
 		const { object, property, modifier, store } = context;
 
 		if (lhs.type === 'Identifier') {
-			lhs = modifier(x`${object}[${property}]`);
-
 			contextual_dependencies.add(object.name);
-			contextual_dependencies.add(property.name);
+			if (property === undefined) {
+				lhs = modifier(b`${object}`[0]);
```

Comment:
This should probably be ``lhs = modifier(x`${object}`)`` but is affected by Rich-Harris/code-red#37